src/trajectory.jl

```diff
@@ -104,6 +105,7 @@ Slice sampler carried during the building of the tree.
 It contains the slice variable and the number of acceptable condidates in the tree.
 """
 struct SliceTreeSampler{F<:AbstractFloat} <: AbstractTreeSampler
```

Comment:
Perhaps rename 

- `SliceTreeSampler` ==> `SliceTrajectorySampler`
- `MultinomialTreeSampler` ==> `MultinomialTrajectorySampler`

since these samplers are for randomly picking a point from a Hamiltonian trajectory -- these trajectories are not always organised in a tree, it could be a static trajectory too (e.g. static HMC with slice/multinomial sampling).

Reply:
Sounds good to me.